timemachine/fe/dummy.py

```diff
@@ -37,10 +36,12 @@ def generate_dummy_group_assignments(
 
     Example
     -------
+    >>> from rdkit import Chem
+    >>> from timemachine.graph_utils import convert_to_nx
     >>> mol = Chem.MolFromSmiles("OC1COO1")
-    >>> g = convert_bond_list_to_nx(get_romol_bonds(mol))
+    >>> bond_graph = convert_to_nx(mol)
     >>> core = [1, 2]
-    >>> list(generate_dummy_group_assignments(g, core))
+    >>> list(generate_dummy_group_assignments(bond_graph, core))
```

Comment:
This function has assertions in it that can trigger failures when calling `get_cores`. Need to be handled if we don't want to bail out of the MCS code.

Seeing failures for `assert len(list(nx.connected_components(bond_graph))) == 1`

Reply:
This should only fail when the bond graph of one of the input mols is disconnected. A difference with this PR is now the failure happens at an earlier stage: during atom mapping instead of at `SingleTopology` setup. But I might argue this is a feature, not a bug?

Reply:
+1 - if bond_graph is disconnected this should probably barf.